### PR TITLE
feat(ajax): add maxCacheSize option

### PR DESCRIPTION
**Pull request by whenlit**
## What I did

1. Add maxCacheSize option to cacheOptions
2. Store size with requests in the cache
3. Add size field to cache
4. Remove old entries from the cache when it is full
5. Added computing of response size from `response.blob().size` when `Content-Length` header is not available

### CODE REVIEW

File: packages/ajax/src/Cache.js [hunk 1]
```diff
@@ -59,7 +106,28 @@ export default class Cache {
     });
   }
 
+  /**
+   * Truncate the cache to the given size, according to a First-In-First-Out (FIFO) policy
+   *
+   * @param {number} maxAllowedCacheSize
+   */
+  truncateTo(maxAllowedCacheSize) {
```
[the remark this answers is not in the document]
Comment (whenlit) [reply packages/ajax/src/Cache.js:114]: Well, the set function doesn't know about the maxSize / newSize / etc. We would have to add it as an argument, or as a constructor argument (which would make sense I guess) but then you couldn't use the maxCacheSize as a fetch option anymore (if you would want that)

File: packages/ajax/src/Cache.js [hunk 2]
```diff
@@ -3,47 +3,94 @@ import './typedef.js';
...
-   * @param {Response} response the cached response
+   * @param {CacheResponse} response the cached response
+   * @param {number} size the response size
    */
-  set(requestId, response) {
+  set(requestId, response, size = 0) {
+    if (this._cachedRequests[requestId]) {
+      this.delete(requestId);
+    }
+
```
[the remark this answers is not in the document]
Comment (whenlit) [reply packages/ajax/src/Cache.js:36]: Ah yes, this is exactly what you're saying here :) I don't know, I kinda like how the cacheManager directs what to do, and the cache decides how it does it.

**Review by LarsDenBakker — APPROVED** [review]
LGTM

File: packages/ajax/src/Cache.js [hunk 3]
```diff
@@ -3,47 +3,94 @@ import './typedef.js';
...
+     * @private
+     */
+    this._size = 0;
+  }
+
+  /**
+   * The size of the cache
+   * @returns {number}
+   */
+  get size() {
```
Comment (narzac) [inline packages/ajax/src/Cache.js:22]: I see with the updated implementation size is no longer have to be part of the public API of Cache class, so  better to keep it "private" in JS context

Comment (whenlit) [reply packages/ajax/src/Cache.js]: Ok, good point! It is done.

**Review by narzac — APPROVED** [review]

**Pull request merged**